wicker/core/datasets.py

```diff
@@ -303,3 +452,85 @@ def dataset_size(self) -> int:
             int: total dataset size in bytes
         """
         return self._get_dataset_partition_size()
+
+
+class DatasetFactory:
+    """Provides functions to make it easier for users to create instances of Wicker datasets by
+    creating them with standardized parameters.
+
+    To customize the dataset creation process, we recommended deriving a new subclass of this
+    factory and overriding its methods (or adding new ones) in order to support the specific kind of
+    Wicker dataset that you want.
+    """
+
+    def create_filesystem_dataset(
+        self,
+        dataset_name: str,
+        dataset_version: str,
+        dataset_partition_name: str,
+        root_path: str,
```

Comment:
Can this root_path config in the Wicker config json?

Reply:
Yes, I updated the `wickerconfig.json` file format to include a `filesystem_config` entry that includes its root path.